ntex-bytes/src/buf/pool.rs

```diff
@@ -0,0 +1,269 @@
+use bytes::{Buf, BufMut, Bytes, BytesMut, buf::UninitSlice};
+use std::future::{self, Future};
+use std::io::IoSlice;
+use std::ops::Deref;
+use std::rc::Rc;
+use std::sync::Arc;
+
+/// A source of buffers.
+///
+/// Dropped buffers will automatically return to this pool.
+pub trait BufferSource {
+    /// The type of buffer returned by this buffer source that owns its contents.
+    type Owned: Owned;
+
+    /// The type of accumulator of buffers used by encoders.
+    type Accumulator: BufferAccumulator<Shared = <Self::Owned as Owned>::Shared>;
+
+    /// The type of future returned by the [`BufferSource::take`] impl.
+    type TakeFuture: Future<Output = Self::Owned>;
+
+    /// Asynchronously returns a buffer that is at least as long as the requested length.
+    fn take(&self, len: usize) -> Self::TakeFuture;
```

Comment:
lets accept `&mut Context` instead of returning future

Reply:
let change pool’s api as well

Reply:
Again, as I explained in https://github.com/ntex-rs/ntex/pull/61#discussion_r743176864 , it is not possible for the pool API to take a `&mut Context` instead of returning `Future`.